subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/declaring_repositories.adoc

```diff
@@ -773,17 +773,19 @@ Gradle provides link:{javadocPath}/org/gradle/api/artifacts/repositories/Authent
 that allows you to declare only the type of required credentials. Credential values are looked up from the <<build_environment.adoc#sec:gradle_configuration_properties,Gradle Properties>>
 during the build that requires them.
 
-For example, given repository configuration:
+For example, given repository configurations:
 
 .Externalized repository credentials
 ====
 include::sample[dir="samples/credentials-handling/publishing-credentials/groovy",files="build.gradle[tags=repositories]"]
 include::sample[dir="samples/credentials-handling/publishing-credentials/kotlin",files="build.gradle.kts[tags=repositories]"]
 ====
 
-The username and password will be looked up from `mySecureRepositoryUsername` and `mySecureRepositoryPassword` properties.
+For both repositories the username and password will be looked up from `mySecureRepositoryUsername` and `mySecureRepositoryPassword` properties.
+
+For the first repository the configuration property prefix - the identity - is determined from the repository name.
+For the second repository the identity is explicitly specified by the user. This is particularly useful when more than one repository uses the same credentials. Note that the identity (provided by either approach) must only have letters or digits in its name.
```

Comment:
The identity name restriction seems a bit artificial. It is enforced by the [CredentialsProviderFactory](https://github.com/gradle/gradle/blob/master/subprojects/model-core/src/main/java/org/gradle/api/internal/provider/CredentialsProviderFactory.java#L158). My understanding is that a dependency repository can have any name, while the publishing repositories are restricted to `[A-Za-z0-9_\-.]+`. Since the properties can include that pattern as well, maybe `CredentialsProviderFactory::validateIdentity` can be extended to match it?

Reply:
You have a good point. I would be tempted to leave it as-is for this PR as the issue is somewhat unrelated to the current PR. I like that you worded the limitation on the identity, very useful.